cmake/onnxruntime_unittests.cmake

```diff
@@ -1848,8 +1849,8 @@ if (WIN32 AND onnxruntime_BUILD_SHARED_LIB AND
                ${ONNXRUNTIME_AUTOEP_LIB_LINK_FLAG})
 
   # test library
-  file(GLOB_RECURSE onnxruntime_autoep_test_SRC "${ONNXRUNTIME_AUTOEP_TEST_SRC_DIR}/*.h"
-                                                "${ONNXRUNTIME_AUTOEP_TEST_SRC_DIR}/*.cc")
+  file(GLOB onnxruntime_autoep_test_SRC "${ONNXRUNTIME_AUTOEP_TEST_SRC_DIR}/*.h"
```

Comment:
Note: `GLOB_RECURSE` compiles the example_plugin_ep files into the test executable, which we don't want to do.

This can cause an issue with how the test executable and the example_ep code want to include `onnxruntime_cxx_api.h`. The former assumes it is linked directly with onnxruntime.dll, but the later wants to be able to set the `OrtApi` object manually.